test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/NuGetPackageDetailsTest.cs

```diff
@@ -0,0 +1,95 @@
+using Xunit;
+
+namespace NuGet.PackageManagement.UI.Test
+{
+    public class NuGetPackageDetailsTest
+    {
+        [Theory]
+        [InlineData("Newtonsoft.Json", "1.2.3")]
+        [InlineData("Moq", "2.3.4")]
+        public void NuGetPackageDetails_BothPropertiesNotNull_NotNull(string name, string version)
+        {
+            //Arrange
+            NuGetVersion nugetVersion;
+            NuGetVersion.TryParse(version, out nugetVersion);
+
+            //Act
+            var packagedetails = new NuGetPackageDetails(name, nugetVersion);
+
+            //Assert
+            string actualPackageName = packagedetails.PackageName;
+            NuGetVersion actualVersion = packagedetails.VersionNumber;
+            Assert.NotNull(packagedetails);
+            Assert.Equal(name, actualPackageName);
+            Assert.Equal(nugetVersion, actualVersion);
+        }
+
+        [Theory]
+        [InlineData(null, "1.2.3")]
+        [InlineData(null, "2.3.4")]
+        public void NuGetPackageDetails_NameNullVersionNotNull_NotNull(string name, string version)
+        {
+            //Arrange
+            NuGetVersion nugetVersion;
+            NuGetVersion.TryParse(version, out nugetVersion);
+
+            //Act
+            var packagedetails = new NuGetPackageDetails(name, nugetVersion);
+
+            //Assert
+            string actualPackageName = packagedetails.PackageName;
+            NuGetVersion actualVersion = packagedetails.VersionNumber;
+            Assert.NotNull(packagedetails);
+            Assert.Null(actualPackageName);
+            Assert.Equal(nugetVersion, actualVersion);
+        }
+
+        [Theory]
+        [InlineData("Newtonsoft.Json", null)]
+        [InlineData("Moq", null)]
+        public void NuGetPackageDetails_NameNotNullVersionNull_NotNull(string name, string version)
+        {
+            //Arrange
+            NuGetVersion nugetVersion;
+            NuGetVersion.TryParse(version, out nugetVersion);
+
+            //Act
+            var packagedetails = new NuGetPackageDetails(name, nugetVersion);
+
+            //Assert
+            string actualPackageName = packagedetails.PackageName;
+            NuGetVersion actualVersion = packagedetails.VersionNumber;
+            Assert.NotNull(packagedetails);
+            Assert.Equal(name, actualPackageName);
+            Assert.Null(actualVersion);
+        }
+
+        [Theory]
+        [InlineData(null, null)]
+        public void NuGetPackageDetails_NameNullVersionNull_NotNull(string name, string version)
+        {
+            //Arrange
+            NuGetVersion nugetVersion;
+            NuGetVersion.TryParse(version, out nugetVersion);
+
+            //Act
+            var packagedetails = new NuGetPackageDetails(name, nugetVersion);
+
+            //Assert
+            string actualPackageName = packagedetails.PackageName;
+            NuGetVersion actualVersion = packagedetails.VersionNumber;
+            Assert.NotNull(packagedetails);
```

Comment:
By definition this is not possible. If you instantiate a new instance of an object, the variable will be pointing to that object and cannot be null. Since you are instantiating this object in the test you're asserting that the new keyword works, which is not necessary.